Scripts/Runtime/Data/Collections/UnsafeTypedStream.cs

```diff
@@ -129,8 +152,20 @@ public UnsafeTypedStream(Allocator allocator, int laneCount) : this(math.max(CHU
         /// Sets the number of lanes to be the maximum amount of worker threads available plus the main thread.
         /// <see cref="JobsUtility.JobWorkerMaximumCount"/> + 1
         /// </summary>
-        /// <param name="allocator">The <see cref="Allocator" /> to use when allocating memory.</param>
-        public UnsafeTypedStream(Allocator allocator) : this(math.max(CHUNK_SIZE / UnsafeUtility.SizeOf<T>(), 1), allocator, ParallelAccessUtil.CollectionSizeForMaxThreads)
+        /// <param name="allocator">The <see cref="Allocator"/> to use when allocating memory for the collection but not the content</param>
+        /// <param name="blockAllocator">The <see cref="Allocator"/> to use when allocating memory for the content in the collection.</param>
+        public UnsafeTypedStream(Allocator allocator, Allocator blockAllocator) : this(ChunkUtil.MaxElementsPerChunk<T>(), allocator, blockAllocator, ParallelAccessUtil.CollectionSizeForMaxThreads)
+        {
+        }
+
+        /// <summary>
+        /// More explicit constructor that allows for specifying how many elements to put into each block. Useful for
+        /// smaller counts so that large blocks aren't allocated if not needed.
+        /// </summary>
+        /// <param name="elementsPerBlock">The number of elements to allocate space for in each block.</param>
+        /// <param name="allocator">The <see cref="Allocator"/> to use when allocating memory for the collection and the content</param>
+        /// <param name="laneCount">The number of lanes to allow reading from/writing to.</param>
+        public UnsafeTypedStream(int elementsPerBlock, Allocator allocator, int laneCount) : this (elementsPerBlock, allocator,allocator, laneCount)
```

Comment:
I played around with it and I can default the allocators but not the elementCount or laneCount because those aren't compile time constants.

But.... because this a struct, I can't have a parameterless constructor so I need the dev to pass in the allocators explicitly like they do on all the Unity built in `NativeCollectionX` so that I can call `this()` with the calculations for elementCount and laneCount.